### PR TITLE
macos getgcc modern is 11

### PR DESCRIPTION
**Pull request by DvirDukhan**

### CODE REVIEW

File: bin/getgcc [hunk 1]
```diff
@@ -69,11 +69,11 @@ class SystemSetup(paella.Setup):
 
     def macos(self):
         if self.modern:
-            self.install("gcc@10")
-            self.run("%s/bin/symlink -f --target /usr/local/bin/gcc-10 --link /usr/local/bin/gcc" % ROOT, sudo=True)
-            self.run("%s/bin/symlink -f --target /usr/local/bin/g++-10 --link /usr/local/bin/g++" % ROOT, sudo=True)
-            self.run("%s/bin/symlink -f --target /usr/local/bin/gcc-10 --link /usr/local/bin/cc" % ROOT, sudo=True)
-            self.run("%s/bin/symlink -f --target /usr/local/bin/g++-10 --link /usr/local/bin/c++" % ROOT, sudo=True)
+            self.install("gcc@11")
```
Comment (chayim) [inline bin/getgcc:72]: Did brew change to no longer have gcc 10, or did you need 11 specifically? IMHO we should just specify a version, and not do this --modern thing. Similar to conda...  WDYT @DvirDukhan

Comment (rafie) [reply bin/getgcc:72]: The `--modern` "thing" goes to say you'd like to get a gcc version beyond what's the native platform/package manager have to offer.

Comment (chayim) [reply bin/getgcc:72]: Right - but you can't specify if you'd like to.  Hence the question.

Comment (rafie) [reply bin/getgcc:72]: The idea here is to have a concept that represent the intersection between minimum need of all modules and availability on different platforms (OSes/distributions/architecture). The resulting version is ever-growing, as platforms support newer version and we find new ways to gain access to newer gcc versions. This saves us from running around modules (and integration branches) and updating explicit/hard-coded versions.
One thing we should notice on macOS is the non-modern version of the current `getgcc` doesn't install gcc at all - the `gcc` command does becomes available but it invokes CLang. I'll fix that - not in this PR.

**Review by rafie — APPROVED** [review]

**Pull request merged**